src/components/MainPage.js

```diff
@@ -152,22 +207,28 @@ class MainPage extends Component<Props, State> {
                         >
                           IMDB score: {item.vote_average}
                         </Typography>
+                        <Typography
+                          gutterBottom
+                          variant="headline"
+                          component="h4"
+                        >
+                          Release date: {item.release_date}
+                        </Typography>
                       </CardContent>
                       <CardActions>
                         <IconButton
                           className={classes.button}
                           aria-label="Delete"
-                          disabled
                           color="primary"
-                          onClick={this.deleteMovieFromFavorites}
+                          onClick={this.deleteMovieFromFavorites(item.id)}

```

Comment:
You creating a new function on every render. Pls, don't 😄 

You can handle such situations in two ways. Frist one - you can make a new component from this `Grid` and pass inside of it all props from `item` and handlers.
Or you can add some `data-*` attribute to this `IconButton` and try to get id from this attribute.

First one - is more "React'ish" and a second - easier to implement.